zenodo/modules/spam/config.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Zenodo.
+# Copyright (C) 2020 CERN.
+#
+# Zenodo is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Zenodo is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Zenodo; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+from invenio_accounts.models import User
+from invenio_accounts.sessions import delete_user_sessions
+from invenio_db import db
+from zenodo.modules.spam.utils import send_spam_user_email, \
+    send_spam_admin_email
+from flask import abort
+
+
+def default_spam_handling(deposit):
+    """Default actions to counter spam detected record."""
+    user = User.query.get(deposit['_deposit']['owners'][0])
+    user.active = False
+    user.confirmed_at = None
+    delete_user_sessions(user)
+    db.session.commit()
+    send_spam_user_email(user.email)
+    send_spam_admin_email(deposit, user)
+    abort(
+        400,
+        ('This deposit has been automatically marked as spam, please contact '
```

Comment:
``Our spam protection system has classified your upload as a potential spam attempt. As a preventive measure and due to significant increase in spam, we have therefore deactivated your user account and logged you out of Zenodo. Your upload has not been published. If you think this is wrong, please contact our support.``